lib/rucio/web/ui/main.py

```diff
@@ -352,6 +361,59 @@ def GET(self):  # pylint:disable=no-self-use,invalid-name
         return check_token(render.subscriptions_editor())
 
 
+class LoadLogfile():
+    """ Loads logfile content list """
+    def GET(self):
+        try:
+            data = web.input()
+            response = requests.get(str(data.file_location), cert=config_get('webui', 'usercert'), verify=False)
+            cont = response.content
+            file_like_object = io.BytesIO(cont)
+            tar = tarfile.open(mode='r:gz', fileobj=file_like_object)
+            jsonResponse = {}
+            for member in tar.getmembers():
+                jsonResponse[member.name] = member.size
+            web.header('Content-Type', 'application/json')
+            return json.dumps(jsonResponse)
+        except Exception:
+            e_type = sys.exc_info()[0]
+            e_value = sys.exc_info()[1]
+            e_traceback = sys.exc_info()[2]
+            return 'Error: ' + str(e_type) + ' ' + str(e_value) + ' ' + str(e_traceback)
```

Comment:
Instead of returning the error as a string, raise an exception with a corresponding HTTP code. Look at the Auth class for an example.